2-extra/3-groceries.js

```diff
@@ -28,13 +28,27 @@ Exercise 1:
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
 
+for (let key in weeklyMealPlan) {
+  for (let i = 0; i < weeklyMealPlan[key].length; i++) {
+    if (weeklyGroceriesToBuy.indexOf(weeklyMealPlan[key][i]) === -1) {
+      weeklyGroceriesToBuy.push(weeklyMealPlan[key][i]);
+    }
+  }
+}
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
-
+for (let key in weeklyMealPlan) {
+  if (key === "saturday" || key === "sunday") {
+    for (let i = 0; i < weeklyMealPlan[key].length; i++) {
+      weekendGroceriesToBuy.push(weeklyMealPlan[key][i]);
+    }
+  }
+}
```

Comment:
The description of the exercise doesn't say it shouldn't contain repeating items, but I think it is implied - since we wouldn't want to write same item twice on a shopping list 😄